.github/workflows/check.yml

```diff
@@ -59,3 +59,36 @@ jobs:
         - name: Run Cargo Tests
           run: |
             cargo test
+  test-coverage:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set-Up
+        run: sudo apt install -y cmake pkg-config libssl-dev git build-essential clang libclang-dev curl
+
+      - name: Install Rustup
+        run: |
+          curl https://sh.rustup.rs -sSf | sh -s -- -y
+          source ~/.cargo/env
+          rustup default stable
+          rustup update nightly
+          rustup update stable
+          rustup target add wasm32-unknown-unknown --toolchain nightly
+          rustup component add llvm-tools-preview
+          cargo install grcov
+
+      - name: Test coverage
+        run: |
+          export CARGO_INCREMENTAL=0
+          export SKIP_BUILD_WASM=true
+          export BUILD_DUMMY_WASM_BINARY=true
+          export LLVM_PROFILE_FILE="llvmcoveragedata-%p-%m.profraw"
+          export WASM_TARGET_DIRECTORY=/tmp/wasm
+          cargo +nightly build
+          export RUSTFLAGS="-Zinstrument-coverage"
+          rm -r target/debug
+          cargo +nightly test --all
```

Comment:
The test coverage workflow contains the commands for automatic testing, so the run-tests workflow can be removed. What do you think?

@irsal

Reply:
For the test coverage workflow, if any of the tests fail, does the job fail?

Reply:
Yes, it will fail.

Reply:
Ok, then we can remove `run-tests` workflow